syntax = "proto3";

import "request.proto";
import "context.proto";

package piscan_pb;

message ClientToServer {
	enum Type {
		GENERAL_REQUEST = 0;
		SCANNER_STATE_REQUEST = 1;
		DEMOD_REQUEST = 2;
		LOCKOUT_REQUEST = 3;

	}

	Type type = 1;
	oneof content {
		GeneralRequest generalRequest = 2;
		ScannerStateRequest scanStateRequest = 3;
		DemodRequest demodRequest = 4;
		LockoutRequest lockoutRequest = 5;
	}
}

message ServerToClient {
	enum Type {
		REQUEST_RESPONSE = 0;
		SCANNER_CONTEXT = 1;
		DEMOD_CONTEXT = 2;
		GENERAL_MESSAGE = 3;
		SIGNAL_LEVEL = 4;
	}

	int32 handle = 1;
	Type type = 2;
	oneof content {
		ScannerContext scannerContext = 3;
		DemodContext demodContext = 4;
		GeneralMessage generalMessage = 5;
		SystemInfo systemInfo = 6;
		SignalLevel signalLevel = 7;
	}
}

